### ISSUE TITLE
False positive for `unneeded_override`

### ISSUE THREAD

**Issue by ppamorim** [realm/SwiftLint#5886]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

If you have the code below present:

```swift
class BaseViewController: UIViewController {
  func loadView(_ addViews: () -> Void) {
    super.loadView()
    addViews()
    view.setNeedsUpdateConstraints()
  }
}

class FooViewController: BaseViewController {
  override func loadView() {
    super.loadView {
      self.configViews()
      self.configTargets()
    }
  }
  //...
}
```

When running the command below, the whole `FooViewController.loadView` is removed, crashing the app.

```bash
$ swiftlint --fix
```

I caught this while doing code review, usually I use `swiftlint --fix` before commiting the code:

![Screenshot 2024-12-05 at 19 52 03](https://github.com/user-attachments/assets/988f9176-27d2-43cb-9bbe-5e2b912a68e1)

### Environment

* SwiftLint version: 0.57.1
* Xcode version: Xcode 15.4
* Installation method used: Homebrew
* Configuration file:

```yml
disabled_rules:
  - identifier_name
  - line_length
  - todo
  - cyclomatic_complexity
  - type_body_length
  - file_length
  - function_body_length
  - function_parameter_count
  - nesting
  - block_based_kvo
  
  # Remove:
  - unused_closure_parameter
  - explicit_type_interface
  - large_tuple
  - inclusive_language

opt_in_rules: # some rules are only opt-in
  - explicit_type_interface
  - force_unwrapping
  - implicit_return
  - attributes
  - closure_end_indentation
  - closure_spacing
  - conditional_returns_on_newline
  - empty_string
  - contains_over_first_not_nil

excluded:
  - Carthage
  - Pods
  - UnitTests
  - UITests

force_cast: warning
force_try: warning
```
